be/src/io/cache/block_file_cache_factory.cpp

```diff
@@ -92,16 +92,20 @@ Status FileCacheFactory::create_file_cache(const std::string& cache_base_path,
     }
     auto cache = std::make_unique<BlockFileCache>(cache_base_path, file_cache_settings);
     RETURN_IF_ERROR(cache->initialize());
-    _path_to_cache[cache_base_path] = cache.get();
-    _caches.push_back(std::move(cache));
+    {
```

Comment:
At the beginnig of this function, the filesystem operations also have concurrency problem? such as:
```
fs->exists
fs->create_directory
fs->delete_directory
fs->create_directory
```

Reply:
the dir path is not same.